backend/src/test/java/com/google/minicrm/data/FormTest.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.minicrm.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.User;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Provides Unit tests for the Form data object.
+ */
+@RunWith(JUnit4.class)
+public final class FormTest {
+
+  private static final String TEST_USER_ID = "testUserId";
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown(){
+    helper.tearDown();
+  }
+
+  @Test
+  public void form_convertToEntityAndBack() throws Exception {
```

Comment:
for Java unit tests, a good naming convention is functionToTest_withCertainInputs_producesCertainResult.

Reply:
Makes sense,
so for testing a method forms.toEntity would a valid name be:
formsToEntity_withValidParameters_producesCorrectEntity?
or should I leave off forms at the beginning and reword a few things?

Also, this currently tests both the toEntity and the new Form(Entity) constructor. Should I leave these tests like these that test multiple functions out of the unit tests?

Reply:
I think this could be something name something like "newForm_fromValidFormAsEntity_producesEquivalentValidForm". Your "Act" step could probably be "Form convertedForm = new Form(form.asEntity)".